### PR TITLE
MRG : Freesurfer tests fix

### PR DESCRIPTION
**Pull request by agramfort**
taking over #305

added hashing to support different versions of fsaverage files.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
You have an error on the old versions of `numpy`:
```
................................................
======================================================================
ERROR: Test IO of .annot
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/nipy/nibabel/venv/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/nipy/nibabel/venv/lib/python2.7/site-packages/numpy/testing/decorators.py", line 146, in skipper_func
    return f(*args, **kwargs)
  File "/home/travis/build/nipy/nibabel/venv/lib/python2.7/site-packages/nibabel/freesurfer/tests/test_io.py", line 105, in test_annot
    labels, ctab, names = read_annot(annot_path)
  File "/home/travis/build/nipy/nibabel/venv/lib/python2.7/site-packages/nibabel/freesurfer/io.py", line 233, in read_annot
    labels[mask] = ord[np.searchsorted(ctab[ord, -1], labels[mask])]
TypeError: array cannot be safely cast to required type
```

File: nibabel/freesurfer/tests/test_io.py [hunk 1]
```diff
@@ -91,7 +110,11 @@ def test_annot():
         if a == 'aparc':
             labels_orig, _, _ = read_annot(annot_path, orig_ids=True)
             np.testing.assert_array_equal(labels == -1, labels_orig == 0)
-            assert_true(np.sum(labels_orig == 0) > 0)
+            # Handle different version of fsaverage
+            if hash_ == 'bf0b488994657435cdddac5f107d21e8':
+                assert_true(np.sum(labels_orig == 0) == 13887)
+            else:
```
Comment (matthew-brett) [inline nibabel/freesurfer/tests/test_io.py]: `elif hash_ == ....` - in case the data changes again?

File: nibabel/freesurfer/tests/test_io.py [hunk 2]
```diff
@@ -1,31 +1,48 @@
...
-if have_freesurfer:
-    subj_dir = os.environ["SUBJECTS_DIR"]
-    subject_id = 'fsaverage'
-    data_path = pjoin(subj_dir, subject_id)
+
+def _hash_file_content(fname):
+    hasher = hashlib.md5()
+    with open(fname, 'rb') as afile:
+        buf = afile.read()
+        hasher.update(buf)
```
Comment (matthew-brett) [inline nibabel/freesurfer/tests/test_io.py]: simpler as `return hashlib.md5(buf).hexdigest()` ?

**Comment by matthew-brett** [comment]
Thanks for this.

Is there any easy way to check against the (stored) output of freesurfer itself?

**Comment by agramfort** [comment]
grrrr !!!!

I tried a few things but I cannot test locally which makes it a nightmare...

anyway to access a python with np 1.5.1 installed? anaconda seems to support 1.6 only...

**Comment by matthew-brett** [comment]
Are you on OSX?

Install Python.org or macports or homebrew Python then install latest pip then `pip install numpy==1.5.1` should get you what you need.

**Comment by agramfort** [comment]
I think I made it !!!

thanks @matthew-brett for the tip

**Comment by agramfort** [comment]
Ready to merge

**Comment by matthew-brett** [comment]
Excellent - thanks.

**Pull request merged**